sql/core/src/test/scala/org/apache/spark/sql/execution/SQLExecutionSuite.scala

```diff
@@ -119,6 +125,37 @@ class SQLExecutionSuite extends SparkFunSuite {
 
     spark.stop()
   }
+
+  test("SPARK-32813: Table scan should work in different thread") {
+    val executor1 = Executors.newSingleThreadExecutor()
+    val executor2 = Executors.newSingleThreadExecutor()
+    SparkSession.cleanupAnyExistingSession()
+
+    withTempDir { tempDir =>
+      try {
+        val tablePath = tempDir.toString + "/table"
+        val df = ThreadUtils.awaitResult(Future {
+          val session = SparkSession.builder().appName("test").master("local[*]").getOrCreate()
```

Comment:
hmm, won't this line set active session?

Reply:
right, it did.

Reply:
oh I see, you use different threads to create the dataframe and execute it.

Reply:
yea, active session was set in first thread, but the second one has no active session.

Reply:
For this case, shall we respect the confs that were used to create the DataFrame? `SparkPlan` actually captures the active session that was used to create it: `SparkPlan.sqlContext`

Reply:
Actually my first comment is to get confs from default session if no active session is found. In this case, default session is set to the active session of first thread.

Reply:
Just use `sqlContext.conf`?

Reply:
Looks better. No notice that we capture it in SparkPlan already.